public/app/plugins/panel/barchart/bars.ts

```diff
@@ -73,30 +70,119 @@ export function getConfig(opts: BarsOptions) {
         const _dir = dir * (ori === 0 ? 1 : -1);
 
         walkTwo(groupWidth, barWidth, sidx - 1, numGroups, barsPerGroup, xDim, null, (ix, x0, wid) => {
-          let lft = Math.round(xOff + (_dir === 1 ? x0 : xDim - x0 - wid));
+          let left = Math.round(xOff + (_dir === 1 ? x0 : xDim - x0 - wid));
           let barWid = Math.round(wid);
+          const canvas = u.root.querySelector<HTMLDivElement>('.u-over');
+          const bbox = canvas?.getBoundingClientRect();
 
           if (dataY[ix] != null) {
             let yPos = valToPosY(dataY[ix]!, scaleY, yDim, yOff);
-
             let btm = Math.round(Math.max(yPos, y0Pos));
             let top = Math.round(Math.min(yPos, y0Pos));
             let barHgt = btm - top;
-
             let strokeWidth = series.width || 0;
 
             if (strokeWidth) {
-              rect(stroke, lft + strokeWidth / 2, top + strokeWidth / 2, barWid - strokeWidth, barHgt - strokeWidth);
+              rect(stroke, left + strokeWidth / 2, top + strokeWidth / 2, barWid - strokeWidth, barHgt - strokeWidth);
             }
 
-            rect(fill, lft, top, barWid, barHgt);
+            rect(fill, left, top, barWid, barHgt);
+
+            let x = ori === ScaleOrientation.Horizontal ? Math.round(left - xOff) : Math.round(top - yOff);
```

Comment:
`ori === ScaleOrientation.Horizontal` changes everything simple comparisons against integers to comparisons against strings pulled from an enum object. :(

can't we just alias 0 and 1 to `HZ` and `VT` constants?

also renaming of variables below throws off a bunch of code alignment. was x/y/w/h really that unclear?

Reply:
ScaleOrientation enum ain't string, it's 0/1

Reply:
> also renaming of variables below throws off a bunch of code alignment. was x/y/w/h really that unclear?


I'm not done with the renames yet ;)